app/frontend/wallet/shelley/types.ts

```diff
@@ -12,9 +12,9 @@ export type TxAux = {
   ttl: number
   certificates: TxCertificate[]
   withdrawals: TxWithdrawal[]
-  auxiliaryData: TxAuxiliaryData
-  auxiliaryDataHash: HexString
-  validityIntervalStart: number
+  auxiliaryData: TxAuxiliaryData | null
+  auxiliaryDataHash: HexString | undefined
```

Comment:
I would rather keep this as `HexString | null` and modify the upstream code to pass such values instead of undefined, so it's clear where such potentially undefined values enter our logic and that we handle them properly, reducing the risk of bugs due to accidentally omitting some parameter only for it to end up as undefined without us noticing